lib/esp32/espnow.toit

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+STATION ::= 0 // Wi-Fi station mode
+SOFTAP  ::= 1 // Not support yet
+
+BROADCAST_MAC ::= #[0xff, 0xff, 0xff, 0xff, 0xff, 0xff]
+
+check_mac_ address/ByteArray:
+  if address.size != 6:
+      throw "ESP-Now MAC address length must be 6 bytes"
+
+check_key_ key/ByteArray:
+  if key.size != 16:
+      throw "ESP-Now key length must be 16 bytes"
+
+check_channel_ channel/int:
+  if channel > 14 or channel < 0:
+      throw "ESP-Now channel range must be 0~14"
+
+class Address:
+  address/ByteArray
+
+  constructor .address:
+    check_mac_ address
+
+  stringify -> string:
+    return "$(%02x address[0]):$(%02x address[1]):$(%02x address[2]):$(%02x address[3]):$(%02x address[4]):$(%02x address[5])"
+
+class Datagram:
+  address/Address
+  data/ByteArray
+
+  constructor .address .data:
+
+class Service:
+  _group ::= ?
+
+  constructor --mode/int=STATION --pmk/ByteArray?=null:
```

Comment:
Any reason you didn't switch to a named constructor, and changed the name of the key parameter?

Reply:
No reason, mainly I don't catch why using named constructor is better, but I can try this.